### PR TITLE
Implement AArch64 and ARMv8 Processor Information

### PR DESCRIPTION
**Pull request by aarongraham9**
Implement AArch64 and ARMv8 Processor Information

Signed-off-by: Aaron Graham <aaron.graham@unb.ca>

### CODE REVIEW

**Comment by aarongraham9** [comment]
Thoughts @knn-k and @0xdaryl?

File: compiler/aarch64/env/ARM64ProcessorEnum.hpp [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+/*
+ * This file will be included within an enum.  Only comments and enumerator
+ * definitions are permitted.
+ */
+
+   ARMv8,
```
Comment (knn-k) [inline compiler/aarch64/env/ARM64ProcessorEnum.hpp]: I would call it `ARMv8_A`.
There are some architectures in the ARMv8 family, and our AArch64 target is ARMv8-A.
ARMv8-M, for example, does not support the AArch64 instruction set.

Comment (aarongraham9) [reply compiler/aarch64/env/ARM64ProcessorEnum.hpp]: Ok. Sounds good!

**Comment by aarongraham9** [comment]
Updated `ARMv8` -> `ARMv8_A` based on @knn-k's suggestion; thanks!

**Comment by 0xdaryl** [comment]
@genie-omr build all

**Pull request merged**